trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/testing/TestingRemoteSession.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.testing;
+
+import com.google.inject.Inject;
+import io.trino.s3.proxy.server.credentials.Credential;
+import io.trino.s3.proxy.server.credentials.Credentials;
+import io.trino.s3.proxy.server.remote.session.RemoteSession;
+import io.trino.s3.proxy.server.remote.session.RemoteSessionConfig;
+import io.trino.s3.proxy.server.remote.session.RemoteSessionController;
+
+import java.util.Optional;
+
+public class TestingRemoteSession
+{
+    private final Credentials sessionCredentials;
+
+    @SuppressWarnings("resource")
+    @Inject
+    public TestingRemoteSession(ManagedS3MockContainer s3MockContainer, RemoteSessionController remoteSessionController, TestingCredentialsController credentialsController)
+    {
+        Credential policyUserCredential = s3MockContainer.policyUserCredential();
+
+        RemoteSessionConfig remoteSessionConfig = new RemoteSessionConfig(policyUserCredential, "us-east-1", "minio-doesnt-care", Optional.empty());
+        RemoteSession remoteSession = remoteSessionController.startRemoteSession(remoteSessionConfig, "ManagedS3MockContainerSession", Optional.empty());
+        sessionCredentials = Credentials.build(policyUserCredential, remoteSession::currentCredential);
+
+        credentialsController.addCredentials(sessionCredentials);
+    }
+
+    public Credentials sessionCredentials()
```

Comment:
If folks want, I can change this to only inject the Credentials with a `@ForSomething` binding annotation. LMK